qiime2/core/type/tests/test_parse.py

```diff
@@ -99,6 +99,22 @@ def test_typevars(self):
         self.assertIs(V1.mapping, W1.mapping)
         self.assertIs(W1.mapping, X1.mapping)
 
+    def test_TypeMap_with_properties(self):
+        inp, _ = TypeMap({
+            C1[Foo % Properties(['A', 'B', 'C'])]: Str,
+            C1[Foo % Properties(['A', 'B'])]: Str,
+            C1[Foo % Properties(['A', 'C'])]: Str,
+            C1[Foo % Properties(['B', 'C'])]: Str,
+            C1[Foo % Properties(['A'])]: Str,
+            C1[Foo % Properties(['B'])]: Str,
+            C1[Foo % Properties(['C'])]: Str,
+        })
+
+        try:
+            inp = ast_to_type(inp.to_ast(), scope={})
```

Comment:
I think a perfect test would be to assert something about the "re-hydrated" type, but this isn't bad either. (You don't really need the try-except as encountering an exception in a test is fairly _unexceptional_.)

Reply:
@ebolyen thanks for the feedback. I modified the test to probe the "re-hydrated", especially mimicking the assertions from the test immediately before this one. Let me know if you think something else should be tested!